lib/init/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+        GROUP_READ
+        GROUP_EXECUTE
+        WORLD_READ
+        WORLD_EXECUTE)
+    else()
+      execute_process(COMMAND /bin/chmod 755 ${GISBASE}/${script_file_name})
+    endif()
+  endif()
+endif()
+]====================]
+
+file(TO_NATIVE_PATH "${OUTDIR}/${GISBASE_DIR}" gisbase_init_dir)
+
+file(COPY ${CMAKE_SOURCE_DIR}/demolocation/PERMANENT
+     DESTINATION ${OUTDIR}/${GRASS_INSTALL_DEMODIR})
+
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/demolocation/PERMANENT/
+        DESTINATION ${GRASS_INSTALL_DEMODIR}/PERMANENT)
+
+set(GISDBASE ${OUTDIR}/${GISBASE_DIR})
+configure_file(${CMAKE_SOURCE_DIR}/demolocation/grassrc.tmpl "${GISRC}" @ONLY)
+unset(GISDBASE)
+
+file(TO_NATIVE_PATH ${OUTDIR}/${GRASS_INSTALL_BINDIR} grass_dll_dir)
+set(DLL_PATH_LIST)
+list(APPEND DLL_PATH_LIST ${grass_dll_dir})
+separate_arguments(DLL_PATH_LIST)
+
+set(LD_LIBRARY_PATH_VAR)
+if(UNIX OR MINGW)
+  if(DEFINED ENV{LD_LIBRARY_PATH})
+    set(LD_LIBRARY_PATH_VAR
+        "${OUTDIR}/${GRASS_INSTALL_LIBDIR}:$ENV{LD_LIBRARY_PATH}")
+  else()
+    set(LD_LIBRARY_PATH_VAR "${OUTDIR}/${GRASS_INSTALL_LIBDIR}")
+  endif()
+endif()
+message(STATUS "GISBASE (build) : ${GISBASE_DIR}")
+message(STATUS "grass config directory (build): ${grass_config_dir}")
+#[[
+if(NOT MSVC)
+  configure_file(env.sh ${grass_config_dir}/bashrc @ONLY)
+  message(STATUS "env variables: ${OUTDIR}/${GRASS_INSTALL_ETCDIR}/config/bashrc")
+endif()
+]]
+
+# configure_file(grass.py ${OUTDIR}/${GRASS_INSTALL_ETCDIR}/${START_UP} @ONLY)
+# message(STATUS "Startup (build): ${GISBASE}/${script_file_name}")
+message(STATUS "gisrc (build): ${GISRC}")
+
+# finish installtree and buildtree configurations
+
+build_program(
+  NAME
+  echo
+  SOURCES
+  echo.c
+  PACKAGE
+  "grasslibs"
+  RUNTIME_OUTPUT_DIR
+  ${GRASS_INSTALL_ETCBINDIR}
+  NO_DOCS)
+
+build_program(
+  NAME
+  run
+  SOURCES
+  run.c
+  PACKAGE
+  "grasslibs"
+  RUNTIME_OUTPUT_DIR
+  ${GRASS_INSTALL_ETCBINDIR}
+  NO_DOCS)
+
+build_program(
+  NAME
+  clean_temp
+  SOURCES
+  clean_temp.c
+  DEPENDS
+  grass_gis
+  PACKAGE
+  "grasslibs"
+  RUNTIME_OUTPUT_DIR
+  ${GRASS_INSTALL_ETCBINDIR}
+  NO_DOCS)
+
+build_program(
+  NAME
+  lock
+  SOURCES
+  lock.c
+  DEPENDS
+  grass_gis
+  grass_datetime
+  PACKAGE
+  "grasslibs"
+  RUNTIME_OUTPUT_DIR
+  ${GRASS_INSTALL_ETCBINDIR}
+  NO_DOCS)
+
+if(MINGW)
+  build_program(
+    NAME
+    winlocale
+    SOURCES
+    winlocale.c
+    RUNTIME_OUTPUT_DIR
+    ${GRASS_INSTALL_ETCBINDIR}
+    NO_DOCS)
+endif()
+
+set(VERSIONNUMBER_CONTENTS "${GRASS_VERSION_NUMBER}")
+if(GRASS_VERSION_GIT)
+  list(APPEND VERSIONNUMBER_CONTENTS
+       "${VERSIONNUMBER_CONTENTS} ${GRASS_VERSION_GIT}")
+endif()
+
+file(WRITE ${OUTDIR}/${GRASS_INSTALL_ETCDIR}/VERSIONNUMBER
+     "${VERSIONNUMBER_CONTENTS}")
+
+install(FILES ${OUTDIR}/${GRASS_INSTALL_ETCDIR}/VERSIONNUMBER
+        DESTINATION ${GRASS_INSTALL_ETCDIR})
+
+execute_process(
+  COMMAND sh ${CMAKE_CURRENT_SOURCE_DIR}/version.sed "${GRASS_VERSION_NUMBER}"
```

Comment:
Is `COMMAND sh` the cmake portable way of doing operations? In other words, does this tie compiling on windows to a msys2/cygwin environment?

Reply:
WIP, Windows will be a challenge…

Reply:
I know. Do you think having the CI ready for windows earlier would help or harm? Like seeing more failures than needed, but also tracking up to where the build can reach.

Reply:
There is bunch of changes needed in the source code. Some already prepared by @rkanavath. It is little chicken and egg problem: you can't compile without the changes, but committing all the changes at once didn't seem good either. Maybe it makes more sense to have Linux CMake merged and then follow with other platforms.